src/system/SystemPacketBuffer.cpp

```diff
@@ -401,6 +401,49 @@ bool PacketBuffer::EnsureReservedSize(uint16_t aReservedSize)
     return true;
 }
 
+uint8_t * PacketBuffer::GetReserve(uint16_t aSize, uint16_t aAlignmentMask)
+{
+    // This private utility method requires that `aAlignmentMask` be one less than a power-of-two alignment. This requirement
+    // is satisfied when aAlignmentMask=alignof(T)-1 for some type T, as passed by the public GetReserve<T>().
+
+    // Computing `reserveStart` can't overflow because a valid PacketBuffer must be larger than kStructureSize.
+    const uintptr_t reserveStart = reinterpret_cast<uintptr_t>(this) + kStructureSize;
+    if (reserveStart + aSize < reserveStart)
+    {
+        // Overflow here means the requested size can't possibly fit.
+        return nullptr;
+    }
+
+    uintptr_t requestStart = (reserveStart + aAlignmentMask) & ~aAlignmentMask;
+    if (requestStart < reserveStart)
+    {
+        // Overflow here means the request can't be satisfied because the alignment is too large.
+        return nullptr;
+    }
+
+    const uintptr_t requestEnd = requestStart + aSize;
+    if (requestEnd < requestStart)
+    {
+        // Overflow here means the requested size can't possibly fit.
+        return nullptr;
+    }
+    if (requestEnd <= reinterpret_cast<uintptr_t>(payload))
```

Comment:
At this point, can we not call `EnsureReservedSize(requestEnd - reserveStart)` instead of reimplementing it?

Reply:
Done, with a change to the test immediately above. (`requestEnd - reserveStart` is the sum of `aSize` and the alignment adjustment, which could exceed 16 bits with suitably nasty arguments.)